### DISCUSSION TITLE
How to `jump to next error`

### DISCUSSION THREAD

**Discussion by nikivdev** [zed-industries/zed#6628] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

I can't find a keybind to `jump to next error` in editor.

**Comment by JosephTLyons** [comment]
Hey @nikitavoloboev, give `f8` and `shift-f8` a whirl.  You can use `esc` once you're done to dismiss the error text in the editor.

Does this work for you?

**Comment by JosephTLyons** [comment]
Just saw you asking for help in #1999, so I'm converting this issue into a discussion.

---

Hmm, that's odd that those commands don't work for you. A few questions. Do you have your macOS function keys set to work as standard function keys? If not, you might have to use the `fn` key in conjunction with `f8` and `shift-f8`.

<img width="718" alt="SCR-20230913-pjql" src="https://github.com/zed-industries/community/assets/19867440/cca6e20c-9412-4aaa-98a8-d93b1e2cc4fb">

If none of that helps, lets try to at least run it through the command prompt to make sure the action works in general and it isn't more than just a key binding issue.

In the command prompt, search for `editor: go to diagnostic` and `editor: go to prev diagnostic`.

Let me know what happens with that info and we can debug further, if need be.

**Reply by nikivdev** [reply]
Hey yea that did actually help, thank you lots.

<img width="436" alt="image" src="https://github.com/zed-industries/community/assets/6391776/193bf33f-3c7d-4524-98a7-6e196727b8c0">

I also rebound it to a key.

On side note, I seriously can't wait until you can make plugins for Zed. It's really a huge deal breaker I can't move my entire workflow of VSCode to Zed due to it. 

Like I want to have full state of the editor programmable. I can't even chain actions inside the keybinds, no `if (state), then action`.

Otherwise, really great work on the editor. ♥️

**Comment by xpe** [comment]
I'm having what is at least a related problem: neither pressing `F8` nor triggering `editor: go to diagnostic` work for me in my Rust project. (This is not for a lack of compilation errors; 🥨 there are many problems in the "Project Diagnostics" view.)

I'm using Zed Preview 0.129.1. I don't think I've made any notable customizations nor installed any extensions.

Are there a few things I should sanity check? Any kind of "troubleshooting guide"?

**Comment by afroozeh** [comment]
I'm having the same issue, neither `F8` works with the default keybinding, nor when choosing a different keyset for this. I'm on MacOS btw.
